atomic_reactor/core.py

```diff
@@ -280,29 +283,131 @@ def export(self, container, chunk_size=1024*2048):
         return retry(self._export, container, chunk_size, retry=self.retry_times)
 
 
-class DockerTasker(LastLogger):
+class ContainerTasker(LastLogger):
     def __init__(self, base_url=None, retry_times=DOCKER_MAX_RETRIES,
                  timeout=120, **kwargs):
+        self.build_method = None
+        self.base_url = base_url
+        self.retry_times = retry_times
+        self.timeout = timeout
+        self._tasker = None
+        self.kwargs = kwargs
+
+        super(ContainerTasker, self).__init__(**kwargs)
+
+    @property
+    def tasker(self):
+        if self._tasker is None:
+            if self.build_method is None:
+                raise AttributeError("Container task type not yet decided")
+
+            if self.build_method in (CONTAINER_IMAGEBUILDER_BUILD_METHOD,
+                                     CONTAINER_DOCKERPY_BUILD_METHOD):
+                self._tasker = DockerTasker(self.base_url, self.retry_times, self.timeout,
+                                            **self.kwargs)
+                logger.info('ContainerTasker will use DockerTasker')
+            else:
+                raise AttributeError('no tasker')
+
+            info, version = self._tasker.get_info(), self._tasker.get_version()
+            logger.debug(json.dumps(info, indent=2))
+            logger.info(json.dumps(version, indent=2))
+
+        return self._tasker
+
+    def build_image_from_path(self, *args, **kwargs):
```

Comment:
Can we have more specific param definitions? I'd prefer having common ABC class with abstractmethods that defines common interface for ContainerTasker, DockerTasker and WhateverTasker.

Reply:
those methods are just basically wrapper methods for methods from specific DockerTasker/PodmanTasker or from CommonTasker
specific param definitions are in there

so when we need to change params for some method, this way we will have to change it just once in specific tasker

Reply:
It may become clearer exactly what methods and arguments we need in this class once we implement PodmanTasker.

Reply:
ok